cloudinit/url_helper.py

```diff
@@ -394,31 +517,94 @@ def wait_for_url(
 
     A value of None for max_wait will retry indefinitely.
     """
-    start_time = time.time()
-
-    def log_status_cb(msg, exc=None):
-        LOG.debug(msg)
 
-    if status_cb is None:
-        status_cb = log_status_cb
+    def default_sleep_time(_, loop_number: int):
+        return int(loop_number / 5) + 1
 
     def timeup(max_wait, start_time):
+        """Check if time is up based on start time and max wait"""
         if max_wait is None:
             return False
         return (max_wait <= 0) or (time.time() - start_time > max_wait)
 
-    loop_n = 0
-    response = None
-    while True:
-        if sleep_time_cb is not None:
-            sleep_time = sleep_time_cb(response, loop_n)
+    def read_url_handle_response(response, url):
```

Comment:
Can we rename this to something like `handle_url_response`? On first glance the name looks like it's involved in calling readurl.